delocate/tools.py

```diff
@@ -345,10 +514,68 @@ def set_install_id(filename: str, install_id: str, ad_hoc_sign: bool = True):
         replace_signature(filename, "-")
 
 
-RPATH_RE = re.compile(r"path (.*) \(offset \d+\)")
+RPATH_RE = re.compile(r"path (?P<rpath>.*) \(offset \d+\)")
+
 
+def _parse_otool_rpaths(stdout: str) -> Dict[str, List[str]]:
+    '''Return the rpaths of the library `filename`.
+
+    Parameters
+    ----------
+    stdout : str
+        The decoded stdout of an 'otool -l' command.
 
-def get_rpaths(filename):
+    Returns
+    -------
+    rpaths : dict of list of str
+        Where the key is the architecture and the value is the list of paths.
+        If the library has no rpaths then the values will be an empty list.
+
+    Examples
+    --------
+    >>> _parse_otool_rpaths("""
```

Comment:
This doctest is already tested.

Reply:
As above.   Yes, we test the doctests, but it is very common for others not to test them.  If the tests were only for CI this would not be an issue.

Also - doctests are the right thing for showing someone how to use a function, but, as here, they can make the docstring verbose and harder to read, if used to do the unit testing.

Reply:
> Yes, we test the doctests, but it is very common for others not to test them. If the tests were only for CI this would not be an issue.

You have to go out of your way to skip these tests.  I've added `--doctest-modules` to the config files, so running `pytest` alone will include them.

> Also - doctests are the right thing for showing someone how to use a function, but, as here, they can make the docstring verbose and harder to read, if used to do the unit testing.

It can make the docstring a lot larger but they're easy to skip past if you don't want to read the examples.  If these examples are hard to read then it's because I didn't always include comments inline with the example.  I don't think there can be too much documentation here.

I always prefer writing doctests to unit tests.  I use unit tests when the functions are too complex to show interactively, like when they have to work with files.  Otherwise I want to combine code examples, regression tests, and coverage into one section as much as possible.

These are non-public functions so I really wanted to keep the tests local rather than in a separate script.  Even if that meant covering exceptions in the doctests.